src/test.data.ts

```diff
@@ -0,0 +1,124 @@
+export const complexObject = {
```

Comment:
Good advice.
I've added two tests that will cover circular structures for arrays and objects.
It will return `false` for validation when something like that is provided to the function.

The escape characters were already handled as part of some strings within the complex object, but I've added some more to improve it's robustness with testing.

I'm not sure how invalid JSON objects and unescaped strings can be added to the tests since they're not very likely to pass all the way to this function successfully. I expect the value passed to this function to be an object that is already in-memory at the runtime, so it should be valid in that term, right? 
Trying to pass something totally broken will not be allowed by JavaScript, TypeScript and all the stuff before the test run. If you have any ideas how that can be achieved, let me know, I can try something.

Reply:
@Mrtenz thoughts on how to test this thoroughly?